infra/src/main/kotlin/fi/fta/geoviite/infra/util/CSVPrintingUtils.kt

```diff
@@ -2,12 +2,22 @@ package fi.fta.geoviite.infra.util
 
 import org.apache.commons.csv.CSVFormat
 import org.apache.commons.csv.CSVPrinter
+import java.time.Instant
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
 
 data class CsvEntry<T>(
     val header: String,
     val getValue: (data: T) -> Any?,
 )
 
+val csvDateFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy")
+fun toCsvDate(date: Instant) = csvDateFormatter
+    // Default to Finnish time. Stored dates are at midnight Finnish time (22:00 UTC previous day),
+    // so a timezone offset is added to make sure the correct date is shown
+    .withZone(ZoneId.of("UTC+2"))
```

Comment:
Hömm onkos tämä totta? Kesällähän Suomen keskiyö on UTC:ssa 21:00. Eli jos siihen lisää vain kaksi tuntia, niin päivä jää yhä edelliseksi.

Mietin (tai kyllä testasinkin) että
```
csvDateFormatter
    .format(LocalDateTime.ofInstant(date, ZoneId.of("Europe/Helsinki")))
```
oisi varmaan korrektimpi, mutta en tiedä että onko olemassa sitten vielä jotain enemmän Kotlinin-makuista asiaa sanoa sama asia.

Reply:
Hyvä pointti 🤔 Kaikki tähän mennessä mulle vastaan tullut tavara oli siten, että luontiaikana oli 22:00. Toisaalta päivämäärä on se mikä meitä ainoastaan tässä kiinnostaa, joten sillä ei ole väliä osuuko tarkka luontiaika täsmälleen keskiyölle vai kellonlyömälle 01:00. Voisin tutkia vielä vähän tarkemmin osaako tuo "Europe/Helsinki" ottaa kesäajan huomioon, mutta voisin muuttaa tuon siihen muotoon tapauksessa kuin tapauksessa

Reply:
Joo tuo Arin sanoma tapa on oikea ja sen pitäisi kyllä huomioida kesä/talviaika oikein. Toki sen ZoneId.of("Europe/Helsinki") voi laittaa myös tuonne withZone -kutsuun sisään. Se zone olisi puhdasoppisempaa muuten ottaa frontilla selaimesta (tulee tuossa samassa muodossa näin: `Intl.DateTimeFormat().resolvedOptions().timeZone`) ja passata queryparametrina.